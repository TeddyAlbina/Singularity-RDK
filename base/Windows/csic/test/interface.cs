using System;
public delegate void E(object sender, string msg);
interface I {
	event E a;
}
class T: I {
	E _x;
	event E I.a {
		add { _x += value; }
		remove { _x -= value; }
	}
	public static event E y;
	public void fire(string s) {
		_x("x", s);
		y("y", s);
	}
	public void F(E handler) {
		((I)this).a += handler;
		y += handler;
	}
}
class U {
	static public void handler(object sender, string msg) {
		Console.WriteLine("{0}: {1}", sender, msg);
	}
	static public void Main() {
		T t = new T();
		((I)t).a += new E(handler);
		t.F(new E(handler));
		t.fire("fire!");
	}
}
