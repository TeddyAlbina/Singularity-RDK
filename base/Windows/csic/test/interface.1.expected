x: fire!
x: fire!
y: fire!
